src/runtime/types/component.d.ts

```diff
@@ -0,0 +1,14 @@
+export type ComponentProps<T> =
```

Comment:
Does these types need to be exported?

Reply:
I made those types internal to this module (84984a99a44d3e5b990559cadaf34e3dbd25e10c), but they are required for `useModal` to infer prop types from the modal to open, and add them to the options.